crates/opencascade/src/section.rs

```diff
@@ -0,0 +1,76 @@
+use crate::primitives::Shape;
+use cxx::UniquePtr;
+use opencascade_sys::ffi;
+
+/// A wrapper around the `BRepAlgoAPI_Section` class.
+pub struct Section {
+    pub(crate) inner: UniquePtr<ffi::BRepAlgoAPI_Section>,
+}
+impl Section {
+    /// Create a new `Section` to intersect `target` by `tool`.
+    pub fn new(target: &Shape, tool: &Shape) -> Section {
+        Section {
+            inner: ffi::BRepAlgoAPI_Section_ctor(
+                target.inner.as_ref().expect("Target was null"),
+                tool.inner.as_ref().expect("Tool was null"),
```

Comment:
I think this should work:

```suggestion
                &target.inner,
                &tool.inner,
```